opennlp-tools/src/main/java/opennlp/tools/postag/POSTaggerME_TS.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package opennlp.tools.postag;
+
+import opennlp.tools.util.Sequence;
+
+/**
+ * A thread-safe version of the POSTaggerME. Using it is completely transparent. You can use it in
+ * a single-threaded context as well, it only incurs a minimal overhead.
+ */
+public class POSTaggerME_TS implements POSTagger {
```

Comment:
Basically, we are going to wrap the original implementation to provide thread-safe variants.

 I am wondering (as discussed in the related Jira) what we would need to change in the original impl to make it thread-safe out of the box. 

From a quick dive in the source code, it looks like we would need to ensure, that the models used can be used concurrently + the components have no internal state. The latter would require some more rewriting, which might break the current api, so it would need to be conducted carefully.

Don't know, if it is worth the effort or if the wrapper impl is enough for the use-case (for now).

Reply:
I'd vote the wrapper is sufficient. I appreciate the work on this.

Reply:
> I'd vote the wrapper is sufficient. I appreciate the work on this.
+1